flink-batch-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormat.java

```diff
@@ -139,13 +163,10 @@ public void close() throws IOException {
 	@Override
 	public boolean reachedEnd() throws IOException {
 		try {
-			if (resultSet.isLast()) {
-				close();
-				return true;
-			}
-			return false;
+			return !resultSet.next();
```

Comment:
"Just call it once" isn't a particularly good argument; i could just as well say "don't try to read from an empty table".

you could an IsConsumed flag to prevent records from being skipped.